### ISSUE TITLE
[rv_core_ibex] Improve V2(S) pass rates

### ISSUE THREAD

**Issue by johngt** [lowRISC/opentitan#18421]
### Description

V2 tests that require fixing to 90% pass rate (passing/total):
- [ ] Riscv_debug_branch_jump_test (8/10)
- [ ] Riscv_dret_test (1/5)
- [ ] Riscv_debug_ebreakmu_test (11/15)
- [ ] Riscv_assorted_traps_interrupts_debug_test (4/10)
- [ ] Riscv_interrupt_wfi_test (12/15)
- [ ] Riscv_pmp_out_of_bounds_test (39/50)
- [ ] Riscv_epmp_mmwp_test (14/20)
V2S tests  that require fixing to 90% pass rate (passing/total):
- [ ] Riscv_pc_intg_test (11/15)
- [ ] Riscv_mem_intg_error_test (12/15)

Estimate from scoping 3-6d
>estimate 24

**Comment by GregAC** [comment]
I don't think we need to improve the pass rates in any of these tests for M2.5.1. There's no failures that look concerning and they generally relate to co-sim/ibex mismatches in some tricky corner cases where there's multiple possible correct behaviours.

**Comment by moidx** [comment]
@GregAC  is planning to triage the coverage issues and determine if this issue can be moved to the backlog as part of the rv_core_ibex V2.5 sign-off.

**Comment by GregAC** [comment]
This will move to the backlog and be complete for V3. The failures here are all thought to be verification environment related. They typically relate to complex cases around interrupts/exceptions/debug requests in particular. Here both Ibex and Spike do spec compliant things but don't always match, further work is needed to the co-simulation environment to bring them into alignment (a common issue is precise timing of when things like interrupts are signalled to spike).

**Comment by moidx** [comment]
@GregAC to document known issues and coverage gaps in DV, and decide if this is something we can land as part of M5.

**Comment by johngt** [comment]
@moidx - this issue has been superseded by https://github.com/lowRISC/opentitan/issues/23542 
Going to close this as the efforts for regressions etc are captured there.

**Comment by GregAC** [comment]
@johngt this issue is aiming to track post M4/V2(S) work so re-opening.

**Comment by vogelpi** [comment]
@GregAC mentioned to have a PR almost ready to bring the pass rates further up. I am thus adjusting the effort left. Please correct the effort spent if you disagree Greg.

**Comment by GregAC** [comment]
A deep dive regression triage of all Ibex regression failures is here: https://github.com/lowRISC/ibex/issues/2187

As part of this work some of the failures have been addressed and the most recent nightly run https://ibex.reports.lowrisc.org/opentitan/ibex-regr-report-04-07-2024.html has a 98.1% pass rate.

**Comment by vogelpi** [comment]
Thanks @GregAC for your report on the deep dive regression triage. I've reviewed it and confirm that the observed issues are not concerning but rather well understood testbench / co-simulation issues which are in most cases tedious to fix. There was one RTL bug identified which isn't critical and for which a workaround exists, see lowRISC/ibex#2186.

**Comment by vogelpi** [comment]
After discussing with @GregAC , we've decided to move this to M7. For M5, all relevant work is done.
